core/drivers/amd/gpio_common.c

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2002-2021 Xilinx, Inc.  All rights reserved.
+ * Copyright (c) 2022 Foundries.io Ltd. (jorge@foundries.io)
```

Comment:
After discussing this with legal there is no copy paste code from the original Versal OP-TEE port. Additionally, the Versal OP-TEE port used drivers from the original Xilinx Linux driver but wasn't exactly copy paste either and did not need a Xilinx copyright either. According to AMD legal department this is only required when code is reused one-for-one. Looking at the other code and using similar ideas doesn't warrant a copyright.

Reply:
@Akshay-Belsare  That is my understanding unless there is one-for-one code reuse.

Reply:
> After discussing this with legal there is no copy paste code from the original Versal OP-TEE port. Additionally, the Versal OP-TEE port used drivers from the original Xilinx Linux driver but wasn't exactly copy paste either and did not need a Xilinx copyright either. According to AMD legal department this is only required when code is reused one-for-one. Looking at the other code and using similar ideas doesn't warrant a copyright.

hi @nathan-menhorn let me check this with Qualcomm and Foundries.io Legal and will come back to you on the next steps.

Reply:
Qualcomm/Foundries legal still reviewing, will let you know as soon as they come back to me.

However, from my side and at a personal level (as the author and under EU legislation) I am going to ask that my name is retained on all files _derived_ from the original ones. Let me know how you want to proceed.

Reply:
so from my end please keep the copyright + author as that is my understanding, **nak without these changes**